src/rgw/driver/d4n/d4n_directory.cc

```diff
@@ -964,7 +964,13 @@ int BlockDirectory::set(const DoutPrefixProvider* dpp, std::vector<CacheBlock>&
   try {
     boost::system::error_code ec;
     boost::redis::generic_response resp;
-    redis_exec(conn, ec, req, resp, y);
+    if(redis_pool==nullptr)
```

Comment:
suggest making the nearest branch the one we prefer/expect, and don't hesitate to write "if (ptr)" and "if( !ptr )"
`if (redis_pool) {
  // use the pool
} else {
  //sorry, no connection pool
...
}
   
`

Reply:
can use [[likely]] or [[unlikely]] if we know which branch is probably going to be taken